### PR TITLE
Inverse Gamma Node

### PR DESCRIPTION
**Pull request by abpolym**
Adds the [inverse gamma](https://en.wikipedia.org/wiki/Inverse-gamma_distribution) node/distribution.

### CODE REVIEW

File: src/rules/gamma_inverse/out.jl [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export rule
+
+# Belief propagation
+# case: inbound messages on edges [α, θ] sum-product
+@rule GammaInverse(:out, Marginalisation) (m_α::Any, m_θ::Any) = GammaInverse(mean(m_α), mean(m_θ))
```
Comment (bvdmitri) [inline src/rules/gamma_inverse/out.jl:5]: This belief-propagation rule is correct only if both arguments are `PointMass` (remember that for belief propagation we generally do not use `Any`, if you spot somewhere most probably it is a mistake. For VMP in contrast `Any` is OK)
```suggestion
@rule GammaInverse(:out, Marginalisation) (m_α::PointMass, m_θ::PointMass) = GammaInverse(mean(m_α), mean(m_θ))
```

**Review by bvdmitri — CHANGES_REQUESTED** [review]
The PR itself is very good, code is nice and clean. Well done! I spotted only one mistake (in the BP rule) and I would like for someone else to double-check all derivations. After that we can merge.

File: src/rules/gamma_inverse/marginals.jl [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export marginalrule
```
Comment (bvdmitri) [inline src/rules/gamma_inverse/marginals.jl:1]: minor but these exports aren't necessary, they are exported automatically (we should remove it from all other places)

**Comment by albertpod** [comment]
Good job @abpolym! Once @bvdmitri comments are addressed, we can merge!

File: src/rules/gamma_inverse/marginals.jl (hunk 2 above)
Comment (abpolym) [reply src/rules/gamma_inverse/marginals.jl]: I've removed both exports for rule and marginalrule

**Comment by abpolym** [comment]
I've added the requested changes 😄

**Review by bvdmitri — APPROVED** [review]
Thanks @abpolym , well done.

**Pull request merged**